Sources/FunctionalKit/FutureType.swift

```diff
@@ -120,8 +144,10 @@ extension FutureType where ParameterType: FutureType {
 	public var joined: Future<ParameterType.ParameterType> {
 		return Future.unfold { done in self.run { $0.run(done) } }
 	}
+}
 
-	public func flatMap <F> (_ transform: @escaping (ParameterType) -> F) -> Future<F.ParameterType> where F: FutureType {
-		return map(transform).joined
-	}
+extension FutureType {
+    public func flatMap <F> (_ transform: @escaping (ParameterType) -> F) -> Future<F.ParameterType> where F: FutureType {
```

Comment:
Why we call flatMap and not bind, like in other types?

Reply:
I actually used `map` and `flatMap` for all types, save for `Optional` and `Array`, for which I used `fmap` and `bind` because there were conflicts with the definitions in the Swift standard library (where these functions use the `throws-rethrows` pattern). In general, I honestly prefer `map` and `flatMap` to `fmap` and `bind` à la Haskell.